### ISSUE TITLE
High speed plotting

### ISSUE THREAD

**Issue by Vasili-Sk** [klonyyy/MCUViewer#20]
Can we use STM logger (STtrace) library for buffered high data rate plotting? i'm also in motor control and sine is barely visible there :)

**Comment by klonyyy** [comment]
Hello, could you elaborate on the topic, or show the problem you're facing? I'm not sure I understand the question.

**Comment by Vasili-Sk** [comment]
STM Studio had special libraries for logging data from inside chip, this way you can synchronise it to end of FOC routine, for example and have clearer data at PWM speed
![image](https://github.com/klonyyy/STMViewer/assets/1153192/0a7b745d-b252-43ab-bf10-ad0472930888)
These are STM libraries. 
![image](https://github.com/klonyyy/STMViewer/assets/1153192/6b2e45f4-b77e-498b-ad3e-059428e340b3)
when you click this little square in stm studio it changes logger mode to buffered, also in settings you can setup how much calls to DumpTrace to be skipped. DumpTrace is a function you need to call from your code that saves data.

**Comment by klonyyy** [comment]
I must admit I wasn't aware of this feature. STMStudio still keeps surprising me. I'd have to research this topic, but right now If you'd like a synchronous way of logging data you should try the Trace Viewer and SWO output. I'll keep in mind to try it out and see how it works/how it is implemented in STMStudio. Thanks!

**Comment by Vasili-Sk** [comment]
yeah i wish but my boards dont have  SWO anymore since i found this thing and always  used it :P

**Comment by klonyyy** [comment]
Got it. In this case I don't think it's possible to synchronize the readings currently. It seems like a very interesting feature, but most probably it will take some time for me research and implement it. Feel free to contribute on this feature if you have some free time, though! :)

**Comment by dzid26** [comment]
There might be lower hanging fruits:
1. Different acquisition sample times per variable would be great 
2. Or a way to quickly disable some variables with e.g. checkboxes or ctrl+select - on the variable list - without the need to modify plots definition - to boost sampling time of other variables. 

Also, it would be good to know print real sampling time and make it red if it is lower than desired. :)

**Comment by cickok** [comment]
Hello there,

this feature is so called "snapshot mode" and it is really useful for exaple PID controller tuning. We (automatization and motor contorl engineers) will be really appreciated if this feature will be implemented :)

BTW this tool is very nice, much more than original STM studio (which is moreover not supported by new elf v5 or something like that).

Good job and best regards.

**Comment by klonyyy** [comment]
I'll give more info on this topic soon, for now I can say it will be integrated in MCUViewer ;)

**Comment by klonyyy** [comment]
The MCUViewer now supports the recorder module [MCUViewer 1.2.0](mcuviewer.com/#downloads)
